src/Spectre.libClustering/Partition.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Partition.h
+* Provides class implementation that simplifies partition vectors.
+*
+Copyright 2017 Dariusz Kuchta
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "Partition.h"
+
+namespace Spectre::libClustering
+{
+	Partition::Partition(gsl::span<unsigned int> partition)
+	{
+		std::map<unsigned int, int> labelsDictionary;
+
+		int currentIntLabel = 1;
+		for (auto label : partition)
+			if (labelsDictionary.find(label) == labelsDictionary.end())
+				labelsDictionary.emplace(label, currentIntLabel++);
+
+		m_Partition.reserve(partition.size());
+		for (int i = 0; i < partition.size(); i++)
+			m_Partition.push_back(labelsDictionary[partition[i]]);
+	}
+
+	const std::vector<unsigned int>& Partition::Get() const
+	{
+		return m_Partition;
+	}
+
+	bool Partition::Compare(const Partition & lhs, const Partition & rhs, double tolerance)
+	{
+		auto lhsPartition = lhs.Get();
+		auto rhsPartition = rhs.Get();
+
+		if (lhsPartition.size() != rhsPartition.size())
+			return false;
+
+		unsigned int matchesCount = 0;
+		size_t length = lhsPartition.size();
+
+		for (int i = 0; i < length; i++)
+			if (lhsPartition.at(i) == rhsPartition.at(i))
```

Comment:
at() also requires instruction jump and return on each call, while using index operator only requires 0-cost address translation